lib/miq_automation_engine/engine/miq_ae_engine/state_var_hash.rb

```diff
@@ -0,0 +1,24 @@
+module MiqAeEngine
+  class StateVarHash < HashWithIndifferentAccess
+    SERIALIZE_KEY = 'binary_blob_id'.freeze
+
+    def encode_with(coder)
+      coder[SERIALIZE_KEY] =
+        blank? ? 0 : BinaryBlob.new.tap { |bb| bb.store_data("YAML", to_h) }.id
```

Comment:
@gmcculloug should this be segregated based on tenant?

Reply:
Considering the BinaryBlob model does not have a tenant column and this value is never exposed external I do not see that being a concern.

Reply:
@gmcculloug One thing we are going to be missing out after this change is the ability to examine what data the user has been using for state vars. Previously these would get logged in the log would it make sense to log the hash as soon as we decode it so we know what the data looks like.

Reply:
Interesting, I had not considered the logging side of things, but that should be easy to add.  I'll try to get that in later today.

Reply:
@mkanoor I pushed changes to log the hash (as a yaml string) and added tests to validate that we log on both successful and failed conditions in the `init_with` method.